File: Jack.Business/FamiliaCrianca.vb
Public Class FamiliaCrianca

    Public Sub New()
        MyBase.new()
    End Sub

    ''' <summary>
    ''' Método para inserir o registro
    ''' </summary>
    ''' <param name="intCrianca">Criança</param>
    ''' <param name="intFamilia">Familia</param>
    ''' <returns>Boolean. Se a operação foi um sucesso, true.</returns>
    ''' <remarks></remarks>
    Public Function Insert(ByVal intFamilia As Integer, intCrianca As Integer) As Boolean
        Dim blnRetorno As Boolean = False
        Dim oDados As Data.FamiliaCrianca = Nothing
        Try
            oDados = New Data.FamiliaCrianca
            blnRetorno = oDados.Insert(intFamilia, intCrianca)

        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno

    End Function


    ''' <summary>
    ''' Método para excluir o registro
    ''' </summary>
    ''' <param name="intFamilia">Familia</param>
    ''' <returns>Boolean. Se a operação foi um sucesso, true.</returns>
    ''' <remarks></remarks>
    Public Function DeleteFamilia(intFamilia As Integer) As Boolean
        Dim blnRetorno As Boolean = False
        Dim oDados As Data.FamiliaCrianca = Nothing
        Try
            oDados = New Data.FamiliaCrianca
            blnRetorno = oDados.DeleteFamilia(intFamilia)

        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function


    ''' <summary>
    ''' Método para excluir o registro
    ''' </summary>
    ''' <param name="intFamilia">Familia</param>
    ''' <param name="intCrianca">Criança</param>
    ''' <returns>Boolean. Se a operação foi um sucesso, true.</returns>
    ''' <remarks></remarks>
    Public Function DeleteCrianca(ByVal intFamilia As Integer, intCrianca As Integer) As Boolean
        Dim blnRetorno As Boolean = False
        Dim oDados As Data.FamiliaCrianca = Nothing
        Try
            oDados = New Data.FamiliaCrianca
            blnRetorno = oDados.DeleteCrianca(intFamilia, intCrianca)

        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function

    Public Function ObterCriancasByFamilia(intFamilia As Integer) As List(Of Model.Criancas)

        Dim lstDados As List(Of Model.Criancas) = Nothing
        Dim oDados As Data.FamiliaCrianca = Nothing
        Try
            oDados = New Data.FamiliaCrianca
            lstDados = oDados.ObterCriancasByFamilia(intFamilia)

        Catch ex As Exception
            lstDados = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstDados

    End Function

    Public Function ObterCriancasByFamiliaWithRep(intFamilia As Integer) As List(Of Model.Criancas)

        Dim lstDados As List(Of Model.Criancas) = Nothing
        Dim oDados As Data.FamiliaCrianca = Nothing
        Try
            oDados = New Data.FamiliaCrianca
            lstDados = oDados.ObterCriancasByFamiliaWithRep(intFamilia)

        Catch ex As Exception
            lstDados = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstDados

    End Function
End Class

File: Jack.Application/FamiliaCrianca.cs
using Jack.DTO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Jack.Application
{
    public class FamiliaCrianca
    {

        public FamiliaCrianca() : base()
        {
        }

        /// <summary>
        /// Método para inserir o registro
        /// </summary>
        /// <param name="intCrianca">Criança</param>
        /// <param name="intFamilia">Familia</param>
        /// <returns>Boolean. Se a operação foi um sucesso, true.</returns>
        /// <remarks></remarks>
        public bool Insert(int intFamilia, int intCrianca)
        {
            bool blnRetorno = false;
            Repository.RepFamiliaCrianca oDados = null;
            try
            {
                oDados = new Repository.RepFamiliaCrianca();
                blnRetorno = oDados.Insert(intFamilia, intCrianca);

            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;

        }


        /// <summary>
        /// Método para excluir o registro
        /// </summary>
        /// <param name="intFamilia">Familia</param>
        /// <returns>Boolean. Se a operação foi um sucesso, true.</returns>
        /// <remarks></remarks>
        public bool DeleteFamilia(int intFamilia)
        {
            bool blnRetorno = false;
            Repository.RepFamiliaCrianca oDados = null;
            try
            {
                oDados = new Repository.RepFamiliaCrianca();
                blnRetorno = oDados.DeleteFamilia(intFamilia);

            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }


        /// <summary>
        /// Método para excluir o registro
        /// </summary>
        /// <param name="intFamilia">Familia</param>
        /// <param name="intCrianca">Criança</param>
        /// <returns>Boolean. Se a operação foi um sucesso, true.</returns>
        /// <remarks></remarks>
        public bool DeleteCrianca(int intFamilia, int intCrianca)
        {
            bool blnRetorno = false;
            Repository.RepFamiliaCrianca oDados = null;
            try
            {
                oDados = new Repository.RepFamiliaCrianca();
                blnRetorno = oDados.DeleteCrianca(intFamilia, intCrianca);

            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }

        public List<DTOCrianca> ObterCriancasByFamilia(int intFamilia)
        {

            List<DTOCrianca> lstDados = null;
            Repository.RepFamiliaCrianca oDados = null;
            try
            {
                oDados = new Repository.RepFamiliaCrianca();
                lstDados = oDados.ObterCriancasByFamilia(intFamilia);

            }
            catch (Exception ex)
            {
                lstDados = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstDados;

        }

        public List<DTOCriancaRepresentante> ObterCriancasByFamiliaWithRep(int intFamilia)
        {

            List<DTOCriancaRepresentante> lstDados = null;
            Repository.RepFamiliaCrianca oDados = null;
            try
            {
                oDados = new Repository.RepFamiliaCrianca();
                lstDados = oDados.ObterCriancasByFamiliaWithRep(intFamilia);

            }
            catch (Exception ex)
            {
                lstDados = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstDados;

        }
    }
}
